static/js/tools/timeline_page.tsx

```diff
@@ -47,6 +48,8 @@ class Page extends Component<PagePropType, PageStateType> {
   }
 
   handleHashChange() {
+    const svPaths = parseStatVarPath()[0];
+    const svIds = parseStatVarPath()[1];
     const placesPromise = parsePlace();
```

Comment:
before call parsePlace and getTriples, can compare if the current places, statsvars are the same as before, if the same, then no need to call them.

Also in case both of them need to be called, can use Promise.All to wait them all finished, then just setState once.

Reply:
Added the comparison before call parsePlace and getTriples.
Any specific reason to use Promise.All ?

Reply:
The way you do here is fine. Since the place and stats var won't change in the same time.